sequencer/src/block2/entry.rs

```diff
@@ -1,45 +1,65 @@
 use super::{Deserialize, Serialize};
 use crate::VmId;
+use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
 use core::fmt;
+use rand_distr::num_traits::FromBytes;
 use std::mem::size_of;
 
+pub(crate) trait TableEntry: Sized {
+    type Word: CanonicalSerialize
+        + CanonicalDeserialize
+        + TryFrom<usize>
+        + TryInto<usize>
+        + Sized
+        + FromBytes;
+    fn max() -> usize;
+
+    fn checked_add_mut(&mut self, rhs: Self) -> Option<()>;
+    fn zero() -> Self;
+    fn one() -> Self;
```

Comment:
My opinion: I don't think these methods should be part of the trait. I do not want to encourage arithmetic directly on `TableEntry`s. If you want arithmetic then convert to `usize`, do the arithmetic`, then convert back to `TableEntry`.

Also: what's the best way to support things like namespace_id that do not support conversion to/from usize or arithmetic? (See discussion in #1012 )

Reply:
punted to #1076